worker/src/main/java/gov/cms/ab2d/worker/config/RoundRobinBlockingQueue.java

```diff
@@ -0,0 +1,357 @@
+                count--;
+                if (categoryQueues.get(entry.getKey()).isEmpty()) {
+                    categoryQueues.remove(entry.getKey());
+                    categories.remove(entry.getKey());
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean containsAll(@NotNull Collection<?> c) {
+        final ReentrantLock lock = this.lock;
+        lock.lock();
+        try {
+            Optional<Boolean> falseFound = c.stream().map(this::contains).filter(res -> !res).findFirst();
+            if (falseFound.isPresent()) {
+                return false;
+            }
+            return true;
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public boolean addAll(@NotNull Collection<? extends E> c) {
+        throw new UnsupportedOperationException("Cannot put an item directly on the queue");
+    }
+
+    @Override
+    public boolean removeAll(@NotNull Collection<?> c) {
+        final ReentrantLock lock = this.lock;
+        lock.lock();
+        try {
+            long falseFound = c.stream().map(this::remove).filter(res -> !res).count();
+            if (falseFound > 0) {
+                return false;
+            }
+            return true;
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public boolean retainAll(@NotNull Collection<?> c) {
+        throw new UnsupportedOperationException("Complicated and probably not needed so unless it is, not implementing");
+    }
+
+    public int size() {
+        return (int) count;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return count == 0;
+    }
+
+    public E take() throws InterruptedException {
+        final ReentrantLock lock = this.lock;
+        lock.lockInterruptibly();
+        try {
+            while (count == 0) {
+                notEmpty.await();
+            }
+            return getNext();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public Object[] toArray() {
+        throw new UnsupportedOperationException("Complicated and probably not needed so unless it is, not implementing");
+    }
+
+    @Override
+    public <T> T[] toArray(T[] a) {
+        throw new UnsupportedOperationException("Complicated and probably not needed so unless it is, not implementing");
+    }
+
+    public String toString() {
+        if (categories.isEmpty()) {
+            return "Empty";
+        }
+        StringBuilder builder = new StringBuilder();
+        for (String category : categories) {
+            builder.append(category).append(" with ").append(categoryQueues.get(category).size()).append(" Futures; ");
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Get the next round robin Object
+     *
+     * @return
+     */
+    private E getNext() {
+        final ReentrantLock lock = this.lock;
+        lock.lock();
```

Comment:
Wonder if we could just declare all methods that need locking `synchronized` for simplicity; the effect appears to be the same.